### PR TITLE
Feat: Send sse on file formatting

### PR DESCRIPTION
**Pull request by mykalmax**

### CODE REVIEW

File: web/server/models.py [hunk 1]
```diff
@@ -28,6 +28,21 @@
 SUPPORTED_EXTENSIONS = {".py", ".sql", ".yaml", ".yml", ".csv"}
 
 
+class EventName(str, enum.Enum):
+    """An enumeration of possible SSE names."""
+
+    PING = "ping"
+    ERRORS = "errors"
+    WARNINGS = "warning"
```
Comment (vchan) [inline web/server/models.py:36]: Should this be warnings?

File: web/server/models.py [hunk 2]
```diff
@@ -554,3 +560,8 @@ class PlanApplyStageTracker(PlanStageTracker):
 
 class PlanCancelStageTracker(PlanStageTracker):
     cancel: t.Optional[PlanStageCancel] = None
+
+
+class FormatFileStatus(BaseModel):
+    path: str
+    status: Status = Status.SUCCESS
```
Comment (vchan) [inline web/server/models.py:567]: Nit: Does a status of INIT mean anything for formatting a file?

File: web/client/src/library/components/editor/EditorFooter.tsx [hunk 3]
```diff
@@ -59,6 +59,14 @@ export default function EditorFooter({ tab }: { tab: EditorTab }): JSX.Element {
           <EditorIndicator.Light ok={tab.isSaved} />
         </EditorIndicator>
       )}
+      {tab.file.isSQL && (
+        <EditorIndicator
+          className="mr-2"
+          text="Format"
```
Comment (vchan) [inline web/client/src/library/components/editor/EditorFooter.tsx:65]: Should this be Formatted?

**Review by vchan — APPROVED** [review]

File: web/server/models.py (hunk 2 above)
Comment (mykalmax) [reply web/server/models.py:567]: It is possible to have `indeterminate` state, where it is not yet clean if file has been formatted not (sinse we are not storing previous state for files)

Comment (vchan) [reply web/server/models.py:565]: Should the default be INIT then?

**Review by vchan — APPROVED** [review]

**Pull request merged**